testcafe/tests/pages/photos-album-model.js

```diff
@@ -0,0 +1,267 @@
+      "Text: You don't have any album yet"
+    )
+  }
+
+  // @param {String} albumName : Name for the new album
+  // @param { number } photoNumber : Number of photos to add to the new album (it will add the first X photos from the timeline)
+  // click on new album button, check the new album page, give a name to album and select photos
+  async addNewAlbum(albumName, photoNumber) {
+    await isExistingAndVisibile(this.toolbarAlbumsList, 'toolbar (album list)')
+    await isExistingAndVisibile(this.btnNewAlbum, 'New album button')
+    await t.click(this.btnNewAlbum)
+    //Check new album page :
+    await t.expect(getPageUrl()).contains('albums/new')
+    await isExistingAndVisibile(this.inputAlbumName, 'Input album Name')
+    await t
+      .expect(this.inputAlbumName.value)
+      .eql('Untitled album')
+      .expect(this.inputAlbumName.focused)
+      .ok('Input album Name is not focus')
+
+    const allPhotosAlbumCount = await this.getPhotosToAddCount(
+      'On create Album page'
+    )
+    await t.expect(allPhotosAlbumCount).eql(t.ctx.allPhotosStartCount) //all photos are displayed
+    await isExistingAndVisibile(this.btnValidateAlbum, 'Create Album Button')
+
+    await t.typeText(this.inputAlbumName, albumName)
+    await this.selectPhotostoAdd(0, photoNumber)
+    await t.click(this.btnValidateAlbum)
+  }
+
+  // @param {Number} indexStart : which photo is the 1st selected
+  // @param {Number} numOfFiles : Number of photo to add (X photos from timeline)
+  async selectPhotostoAdd(indexStart, numOfFiles) {
+    console.log(`Selecting ${numOfFiles} picture(s) from index ${indexStart}`)
+    for (let i = indexStart; i < indexStart + numOfFiles; i++) {
+      await isExistingAndVisibile(this.photoThumb(i), `${i + 1}th Photo thumb`)
+      await t.click(this.photoCheckbox.nth(i))
+    }
+  }
+
+  // @param {String} albumName : Name of the  album
+  // @param { number } photoNumber : Number of photos expected in this album
+  async checkAlbumPage(albumName, photoNumber) {
+    await this.waitForLoading()
+    await t
+      .expect(getPageUrl())
+      .contains('albums')
+      .expect(getPageUrl())
+      .notContains('new')
+      .expect(this.albumTitle.innerText)
+      .eql(albumName)
+
+    const allPhotosAlbumCount = await photoPage.getPhotosCount('On Album page')
+    await t.expect(allPhotosAlbumCount).eql(photoNumber) //all expected photos are displayed
+  }
+
+  // @param {String} AlbumName : Name of the album
+  async goToAlbum(albumName) {
+    await isExistingAndVisibile(this.album(albumName), albumName)
+    await t.click(this.album(albumName))
+    await this.waitForLoading()
+  }
+
+  // @param {String} OldAlbumName : Name of the album (before renamming)
+  // @param {String} NewAlbumName : New Name of the album (after renamming)
+  // @param {string} exit : enter or click : Choose the exit method
+  async renameAlbum(OldAlbumName, NewAlbumName, exit) {
+    const isEnter = exit === 'enter' ? true : false
```

Comment:
Why not pass `true` or `false` as a parameter instead of a string that we *then* convert to a boolean?

Reply:
I think it's easier to read when using the method : 
``` 
  await photoAlbumPage.renameAlbum(
    ALBUM_DATE_TIME, `New_${ALBUM_DATE_TIME}`, 'enter'
  )

  await photoAlbumPage.renameAlbum(
    ALBUM_DATE_TIME, `New_${ALBUM_DATE_TIME}`, 'click'
  )
```

Reply:
In that case I would suggest something like

```
await photoAlbumPage.renameAlbum(
    ALBUM_DATE_TIME, `New_${ALBUM_DATE_TIME}`, { exit: true }
  )
```

It's too easy to have typos in strings and things go completely unnoticed :|